examples/features/torch_batch_process/batch_inference/generate_embedding/bert_embedding_generation.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import os
+import pathlib
+
+import chromadb
+import torch
+from datasets import load_dataset
+from transformers import BertModel, BertTokenizer
+
+from determined.pytorch import experimental
+
+
+class EmbeddingProcessor(experimental.TorchBatchProcessor):
+    def __init__(self, context):
+        self.tokenizer = BertTokenizer.from_pretrained("bert-base-uncased")
+        self.model = BertModel.from_pretrained("bert-base-uncased", output_hidden_states=True)
+
+        self.device = context.device
+
+        self.model = context.prepare_model_for_inference(self.model)
+
+        self.output = []
+
+        self.context = context
+        self.last_index = 0
+        self.rank = self.context.distributed.get_rank()
+
+        self.output_dir = "/tmp/data/embeddings"
+        os.makedirs(self.output_dir, exist_ok=True)
+
+    def process_batch(self, batch, batch_idx) -> None:
+        with torch.no_grad():
+            tokenized_text = self.tokenizer.batch_encode_plus(
+                batch["text"],
+                truncation=True,
+                padding="max_length",
+                max_length=512,
+                add_special_tokens=True,
+            )
+            inputs = torch.tensor(tokenized_text["input_ids"])
+            inputs = inputs.to(self.device)
+            masks = torch.tensor(tokenized_text["attention_mask"])
+            masks = masks.to(self.device)
+
+            outputs = self.model(inputs, masks)
+            outputs = torch.mean(outputs["hidden_states"][-1], dim=1)
```

Comment:
I'd add here some text explaining how embeddings are built since we are using one of many simple strategies. For instance: To create an embedding vector for each document, first, we take the hidden states from the last layer (output["hidden_states"][-1]), which results in a tensor of (#examples, #tokens, #hidden_states) size. Next, we calculate the average across the token-dimension, resulting in (#examples, #hidden_states) vector.